save-demo/src/main/kotlin/com/saveourtool/save/demo/utils/HttpUtils.kt

```diff
@@ -0,0 +1,21 @@
+@file:Suppress("HEADER_MISSING_IN_NON_SINGLE_CLASS_FILE")
+
+package com.saveourtool.save.demo.utils
+
+import io.ktor.utils.io.*
+import io.ktor.utils.io.jvm.javaio.*
+import org.springframework.core.io.buffer.DataBufferUtils
+import org.springframework.core.io.buffer.DefaultDataBufferFactory
+import reactor.core.publisher.Flux
+import java.nio.ByteBuffer
+
+private const val DEFAULT_BUFFER_SIZE = 4096
+
+/**
+ * @return [Flux] of [ByteBuffer]s read from [ByteReadChannel]
```

Comment:
What happens if

 - the read channel is empty?
 - an `IOException` is thrown when reading from the channel?
 
 Please document this behaviour.

Reply:
Oh ok

Reply:
For now I am not that sure about the behavior you asked to consider. Opened #1528 in order to add unit tests for this logic.

Reply:
@0x6675636b796f75676974687562 thanks for your commitment